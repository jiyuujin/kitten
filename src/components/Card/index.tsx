import React, { FC } from 'react'
import { useIntl } from 'gatsby-plugin-react-intl'
import * as SC from './index.module.scss'
import LazyImage from '../LazyImage'
import { SiteMetadataTypes } from '../../types'
import { NekoButton } from 'nekohack-ui'
import PaperPlaneSvg from '../../static/icons/paper-plane.svg'

const Card: FC<any> = ({ data }: SiteMetadataTypes) => {
  const intl = useIntl()

  return (
    <div className={SC.content}>
      <div className={SC.cardInfo}>
        <div className={SC.me}>
          <LazyImage assetUrl={'me.png'} alt={'me'} />
        </div>
        <h1 className={SC.name}>{data.author}</h1>
        <ul className={SC.position}>
          {data.position.split(',').map((p: string) => (
            <li key={p}>{p}</li>
          ))}
        </ul>
        <p className={SC.contact}>
          <NekoButton
            onClick={() => {
              const url =
                                'https://docs.google.com/forms/d/e/1FAIpQLSfFzwsCVnFbu-lV0Jz2fEYHR97hpBSK2g6kEwc-G1fo976ngA/viewform'
              window.open(url, '_blank')
            }}
          >
            {intl.formatMessage({ id: 'contact' })}
            <PaperPlaneSvg
              style={{
                width: '20px',
                height: '20px',
                marginLeft: '4px',
              }}
            />
          </NekoButton>
        </p>
      </div>
    </div>
  )
}

export default Card
