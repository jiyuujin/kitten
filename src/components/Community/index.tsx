import React, { FC } from 'react'
import { useIntl } from 'gatsby-plugin-react-intl'
import * as SC from './index.module.scss'

type CommunityProps = {
    data: Array<unknown>
}

const Community: FC<CommunityProps> = ({ data }) => {
  const intl = useIntl()

  return (
    <div className="wrapper">
      <div className={`section ${SC.section_community}`}>
        <div className={SC.subtitle}>
          {intl.formatMessage({ id: 'community' })}
        </div>
        <div className={SC.description}>
          <ul>
            {data.map(({ node }: any) => {
              return (
                <li key={node.id}>
                  <a
                    href={node.url}
                    role="button"
                    aria-pressed="true"
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    {node.text}
                  </a>
                </li>
              )
            })}
          </ul>
        </div>
      </div>
    </div>
  )
}

export default Community
