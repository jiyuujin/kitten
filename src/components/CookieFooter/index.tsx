import React, { FC } from 'react'
import * as SC from './index.module.scss'
import { getCookie, setCookie } from '../../services/cookie'

export const CookieFooter: FC = () => {
    const accept = () => {
        const isAccepted: boolean = true
        setCookie(isAccepted)
    }

    return (
        <>
            {!getCookie() ? (
                <footer className={SC.cookieFooter}>
                    <div className={SC.content}>
                        <div className={SC.title}>
                            あなたがこのサイト上でベストな体験ができるようCookieを使用しています
                        </div>
                        <a
                            href="https://nekohack-privacy-policy.netlify.app"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            プライバシーポリシー
                        </a>
                        <button onClick={accept}>確認しました</button>
                    </div>
                </footer>
            ) : null}
        </>
    )
}
