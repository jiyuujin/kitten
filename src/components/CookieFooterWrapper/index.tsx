import React, { FC } from 'react'
import { applyPolyfills, defineCustomElements } from '@nekohack/tiny-components/loader'

export const CookieFooterWrapper: FC = () => {
    applyPolyfills().then(() => {
        defineCustomElements()
    })

    return (
        <>
            {/*@ts-ignore*/}
            <cookie-footer
                name="yuukit-me-profile-site"
                title="あなたがこのサイト上でベストな体験ができるようCookieを使用しています"
                url="https://nekohack-privacy-policy.netlify.app"
            />
        </>
    )
}
