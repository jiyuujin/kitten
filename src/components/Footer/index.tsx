import React, { FC } from 'react'
import { useIntl } from 'gatsby-plugin-react-intl'
import * as SC from './index.module.scss'

const Footer: FC = () => {
  const intl = useIntl()

  return (
    <footer className={SC.footer}>
      <p className={SC.contact}>
        <a
          href="https://docs.google.com/forms/d/e/1FAIpQLSfFzwsCVnFbu-lV0Jz2fEYHR97hpBSK2g6kEwc-G1fo976ngA/viewform"
          target="_blank"
          rel="noopener noreferrer"
        >
          {intl.formatMessage({ id: 'contact' })}
        </a>
      </p>
      <p className={SC.copyright}>Created Â© 2020 jiyuujin LAB. All Rights Reserved.</p>
    </footer>
  )
}

export default Footer
