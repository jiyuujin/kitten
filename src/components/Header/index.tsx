import React, { FC } from 'react'
import * as SC from './index.module.scss'

type ProfileHeader = {
    title: string
    author: string
    locale: string
}

export const Header: FC<ProfileHeader> = ({ title, author, locale }) => {
    return (
        <div className={SC.title}>
            {title}
            <div className={SC.name}>
                {author}
            </div>
            <div className={SC.locale}>
                {locale}
            </div>
        </div>
    )
}
