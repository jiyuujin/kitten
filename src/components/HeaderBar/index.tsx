import React, { FC } from 'react'
import { IntlContextConsumer, changeLocale } from 'gatsby-plugin-intl'
import * as SC from './index.module.scss'

const languageName: {
    [field: string]: string
} = {
    en: 'English',
    ja: '日本語',
}

const HeaderBar: FC = () => {
    return (
        <div className={SC.headerBar}>
            <IntlContextConsumer>
                {({
                    languages,
                    language: currentLocale,
                }: {
                    languages: string[]
                    language: string
                }) =>
                    languages.map((language) => (
                        <a
                            key={language}
                            onClick={() => changeLocale(language)}
                            style={{
                                color:
                                    currentLocale === language
                                        ? 'yellow'
                                        : 'white',
                                margin: 10,
                                textDecoration: 'underline',
                                cursor: 'pointer',
                            }}
                        >
                            {languageName[language]}
                        </a>
                    ))
                }
            </IntlContextConsumer>
        </div>
    )
}

export default HeaderBar
