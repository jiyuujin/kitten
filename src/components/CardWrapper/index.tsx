import React, { FC, useState } from 'react'
import { useIntl } from 'gatsby-plugin-intl'
import { NekoButton } from 'nekohack-ui'
import Modal from '../Modal'
import Card from '../Card'

import IdCardSvg from '../../static/icons/id-card.svg'

type CardWrapperProps = {
    data: GatsbyTypes.Maybe<
        Pick<
            GatsbyTypes.SiteSiteMetadata,
            | 'title'
            | 'description'
            | 'biography'
            | 'position'
            | 'workExperience'
            | 'author'
            | 'locale'
            | 'lonlat'
            | 'siteUrl'
            | 'image'
            | 'blogName'
        >
    >
}

export const CardWrapper: FC<CardWrapperProps> = ({ data }) => {
    const intl = useIntl()

    const [open, setOpen] = useState(false)

    const handleOpen = () => {
        setOpen(!open)
    }

    return (
        <>
            <NekoButton onClick={handleOpen}>
                <IdCardSvg
                    style={{
                        width: '20px',
                        height: '20px',
                        marginRight: '4px',
                    }}
                />
                {intl.formatMessage({ id: 'online_business_card' })}
            </NekoButton>
            {open && (
                <Modal onClose={() => setOpen(!open)}>
                    <Card data={data} />
                </Modal>
            )}
        </>
    )
}
