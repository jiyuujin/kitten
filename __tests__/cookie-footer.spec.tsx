import React from 'react'
import renderer from 'react-test-renderer'
import { CookieFooterWrapper } from '../src/components/CookieFooterWrapper'

describe('Cookie Footer Wrapper', () => {
    it('render correctly', () => {
        const tree = renderer.create(<CookieFooterWrapper />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
