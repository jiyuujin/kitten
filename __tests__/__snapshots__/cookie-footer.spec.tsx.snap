// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Footer Wrapper render correctly 1`] = `
<cookie-footer
  name="yuukit-me-profile-site"
  title="あなたがこのサイト上でベストな体験ができるようCookieを使用しています"
  url="https://nekohack-privacy-policy.netlify.app"
/>
`;
